### PR TITLE
fix: reject empty checkout orders

### PR DESCRIPTION
**Pull request by swapnil-turing**
## Summary
- reject checkout requests when cart_items is empty
- reject duplicate cart item ids so request semantics are explicit
- strengthen regression coverage to assert cart_items validation errors

## Notes
- scope is checkout serializer validation and related API tests only

### CODE REVIEW

File: augment-store/server/checkout/serializers.py [hunk 1]
```diff
@@ -92,8 +92,10 @@ class Meta:
 
     def validate_cart_items(self, value):
         user = self.context.get("request").user
+        if not value:
+            raise serializers.ValidationError("cart_items cannot be empty")
         cart_items = CartItem.objects.get_user_cart_items(user).filter(id__in=value)
-        if cart_items.count() != len(value):
+        if cart_items.count() != len(set(value)):
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/checkout/serializers.py]: value:useful; category:bug; feedback:Chose useful because this comment surfaced an ambiguous input-case that can hide request intent in a core checkout API. It identified that duplicate cart item IDs could be silently collapsed, which risks creating orders that do not clearly correspond to what clients submitted.

File: augment-store/server/checkout/tests.py [hunk 2]
```diff
@@ -103,12 +103,11 @@ def test_create_order_with_empty_cart_items(self):
         }
         response = self.member_client.post(url, payload, format='json')
 
-        # THEN we should get a 201 response (order created but empty)
-        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        # THEN we should get a 400 response
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/checkout/tests.py:107]: value:good-to-have; category:bug; feedback:Marked good-to-have because it improved the precision of the regression signal for validation behavior. It highlighted that a generic 400 assertion could pass for unrelated failures and miss a break in cart_items-specific validation.

File: augment-store/server/checkout/serializers.py [hunk 3]
```diff
@@ -92,8 +92,10 @@ class Meta:
 
     def validate_cart_items(self, value):
         user = self.context.get("request").user
+        if not value:
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/checkout/serializers.py:95]: value:good-to-have; category:documentation; feedback:Chose good-to-have because it improved review accuracy by aligning change metadata with the actual code scope. It identified a title and description mismatch that could mislead reviewers and release tracking about which subsystem was modified.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/checkout/serializers.py [hunk 4]
```diff
@@ -92,6 +92,10 @@ class Meta:
...
+            raise serializers.ValidationError("cart_items cannot be empty")
+        if len(value) != len(set(value)):
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/checkout/serializers.py:97]: value:good-to-have; category:bug; feedback:Marked good-to-have because this comment improved regression completeness for an edge-case validation path. It identified that duplicate cart item inputs could lose explicit test coverage, making future behavior drift harder to catch before merge.

**Comment by swapnil-turing** [comment]
augment review

**Review by michaelos443 — APPROVED** [review]
